### PR TITLE
Break out create toolbox logic from addDatabase

### PR DESCRIPTION
**Pull request by flip111**
For https://github.com/gabordemooij/redbean/issues/813

### CODE REVIEW

**Comment by flip111** [comment]
You can call the `R::createToolbox` function to create a toolbox for you. This function has a similar interface as `R::setup` but without side effects: it will only create the toolbox and return it.

    R::createToolbox( 'mysql:host=localhost;dbname=mydatabase', 'myusername', 'mypassword' );

You also can manually assemble your toolbox like this:

    $pdo = new RPDO( $dsn );
    $adapter = new DBAdapter( $pdo );
    $writer = new MySQL( $adapter );
    $oodb = new OODB( $writer );
    $tb = new ToolBox( $oodb, $adapter, $writer );

**Comment by flip111** [comment]
@gabordemooij this comment https://github.com/gabordemooij/redbean/pull/814#issuecomment-614537023 is supposed to replace the documentation on the website

File: RedBeanPHP/Facade.php [hunk 1]
```diff
@@ -398,11 +398,49 @@ public static function addDatabase( $key, $dsn, $user = NULL, $pass = NULL, $fro
 			throw new RedException( 'A database has already been specified for this key.' );
 		}
 
+		self::$toolboxes[$key] = self::createToolbox($key, $dsn, $user, $pass, $frozen, $partialBeans);
```
Comment (Lynesth) [inline RedBeanPHP/Facade.php:401]: This should be
`self::$toolboxes[$key] = self::createToolbox($dsn, $user, $pass, $frozen, $partialBeans);`
You have one too many argument.

File: RedBeanPHP/Facade.php [hunk 2]
```diff
@@ -398,11 +398,49 @@ public static function addDatabase( $key, $dsn, $user = NULL, $pass = NULL, $fro
...
+	 * a connection is really required, for instance when attempting to load a bean.
+	 *
+	 * @param string  $dsn      Database connection string
+	 * @param string  $username Username for database
+	 * @param string  $password Password for database
+	 * @param boolean $frozen   TRUE if you want to setup in frozen mode
+	 *
+	 * @return ToolBox
+	 */
+  public static function createToolbox( $dsn = NULL, $username = NULL, $password = NULL, $frozen = FALSE, $partialBeans = FALSE )
```
Comment (Lynesth) [inline RedBeanPHP/Facade.php:437]: Couldn't this be actually moved into the Toolbox constructor ? Would it make more sense ?

File: RedBeanPHP/Facade.php (hunk 1 above)
Comment (flip111) [reply RedBeanPHP/Facade.php:401]: I changed this in https://github.com/gabordemooij/redbean/pull/815

**Pull request merged**